### DISCUSSION TITLE
Add Or Change Identity Claims In .Net 8 Blazor Web App

### DISCUSSION THREAD

**Discussion by MahdiElahi** [dotnet/aspnetcore#53902] [General]
Hi, In Create New Blazor Web App Project : Check => Individual 

Check => .Net 8 

Check => Auto Interactivew

and in ApplicationUser.cs I add a property for use full name 
```
{

public sting Fullname { get; set; }

}
```

but in components i want show current user full name , but how to do it ? how to add custom claims to user and get in view component for example : i want in component use like this:

  ```
<AuthorizeView>

@Hello @context.User.Identity.Fullname

  </AuthorizeView>
```

**Comment by teacher-zhou** [comment]
Declare a static method for `IIdentity` interface and get the value from specified type

The example code like:
```cs
public static string? GetFullName(this IIdentity identity)
  => (identity as ClaimIdentity)?.FindClaim("FullName")?.ToString();
```

and then you can get
```html
<AuthorizeView>

@Hello @context.User.Identity.GetFullName()

</AuthorizeView>
```

**Comment by dotnet-policy-service** [comment]
Looks like this PR hasn't been active for some time and the codebase could have been changed in the meantime.
To make sure no conflicting changes have occurred, please rerun validation before merging. You can do this by leaving an `/azp run` comment here (requires commit rights), or by simply closing and reopening.

**Comment by dotnet-policy-service** [comment]
Looks like this PR hasn't been active for some time and the codebase could have been changed in the meantime.
To make sure no conflicting changes have occurred, please rerun validation before merging. You can do this by leaving an `/azp run` comment here (requires commit rights), or by simply closing and reopening.
